### PR TITLE
Ignore changes to TKGDefaultCompatibilityImagePath in verify script

### LINKED ISSUE #1270: Ignore changes to TKGDefaultCompatibilityImagePath in verify script
**Bug description**

`usebom` directive during tests results in a diff and the verify script fails. The script should ignore changes to [this line](https://github.com/vmware-tanzu/tanzu-framework/blob/5338922f196a8cbb33114df45bf9f725c5f9579c/pkg/v1/tkg/tkgconfigpaths/zz_bundled_default_bom_files_configdata.go#L14) to unblock tests.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [x] Test and Release
- [ ] User Experience

**Expected behavior**

`usebom` directive should not result in `verify.sh` script failing.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by rajathagasthya**
### What this PR does / why we need it

Ignore changes to TKGDefaultCompatibilityImagePath in verify script to support `usebom` directive in tests.

### Which issue(s) this PR fixes

Fixes #1270 

### Describe testing done for PR

Change [this line](https://github.com/vmware-tanzu/tanzu-framework/blob/5338922f196a8cbb33114df45bf9f725c5f9579c/pkg/v1/tkg/tkgconfigpaths/zz_bundled_default_bom_files_configdata.go#L14) and run `hack/verify-dirty.sh`. The script does not return an error.

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by rajathagasthya** [comment]
//usebom:sandbox/v1.6.0-zshippable/383947424286145198/tkg-compatibility:v2

File: hack/verify-dirty.sh [hunk 1]
```diff
@@ -24,7 +29,7 @@ echo "#############################"
 echo "Verify make configure-bom..."
 echo "#############################"
 make configure-bom
-if ! (git diff --quiet HEAD -- .); then
+if ! (git diff --quiet HEAD -- . "${ignore_file}"); then
```
Comment (vuil) [inline hack/verify-dirty.sh:32]: given $ignore_file is literally the only file configure-bom touches, this renders this part of the check useless. But we can improve on this in a followup.

**Review by vuil — APPROVED** [review]
lgtm, thanks for the fix

File: hack/verify-dirty.sh (hunk 1 above)
Comment (rajathagasthya) [reply hack/verify-dirty.sh:32]: Is the `--ignore-matching-lines` option a solution to this? `usebom` will always change this file, so this check will always fail, right?

**Pull request merged**